homeassistant/components/goalzero/manifest.json

```
@@ -4,6 +4,9 @@
   "config_flow": true,
   "documentation": "https://www.home-assistant.io/integrations/goalzero",
   "requirements": ["goalzero==0.1.7"],
+  "dhcp": [
+    {"hostname": "yeti*", "macaddress": "A4CF12*"}
```

```suggestion
    {"hostname": "yeti*"}
```

This is unique enough that you can drop the oui and implement `async_step_dhcp` in the config flow to abort in the very rare case (probably won't happen unless someone is very creative) you match a non-goalzero device.

You should also implement in `async_step_dhcp` likely in this order

1. Set the unique id (likely the mac)
2. Abort if unique id already configured
3. Abort if the specific host is already configured
4. Abort if its a non-goalzero device
5. pre-filling the ip in the form and showing the form

---

Or even better, don't show the IP in the form at all, as it is known at that point.

---

Since no auth is required, you could also do `self._set_confirm_only()`